src/Uno.UWP/Storage/Streams/DataReader.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.Buffers;
+using System.Text;
+using Uno.Disposables;
+
+namespace Windows.Storage.Streams
+{
+	public sealed partial class DataReader : IDataReader, IDisposable
+	{
+		private readonly static ArrayPool<byte> _pool = ArrayPool<byte>.Create();
+		private readonly IBuffer _buffer;
+
+		private int _bufferPosition = 0;
+
+		private DataReader(IBuffer buffer)
+		{
+			_buffer = buffer;
+		}
+
+		public static DataReader FromBuffer(IBuffer buffer) => new DataReader(buffer);
+
+		public UnicodeEncoding UnicodeEncoding { get; set; }
+
+		public ByteOrder ByteOrder { get; set; }
+
+		public uint UnconsumedBufferLength => (uint)(_buffer.Length - _bufferPosition);
+
+		public byte ReadByte()
+		{
+			VerifyRead(1);
+			return ReadByteFromBuffer();
+		}
+
+		public void ReadBytes(byte[] value)
+		{
+			VerifyRead(value.Length);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'value' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)